### ISSUE TITLE
ProGuard and Orientation annotation

### ISSUE THREAD

**Issue by Krisztiaan** [BelooS/ChipsLayoutManager#31]
dependency loaded from jCenter, ProGuard complains that it can not find Orientation

**Comment by icarovirtual** [comment]
I also found these warning, the logs are below:
```
Warning:com.beloo.widget.chipslayoutmanager.ChipsLayoutManager: can't find referenced class com.beloo.widget.chipslayoutmanager.Orientation
Warning:com.beloo.widget.chipslayoutmanager.ChipsLayoutManager$Builder: can't find referenced class com.beloo.widget.chipslayoutmanager.Orientation
Warning:com.beloo.widget.chipslayoutmanager.IChipsLayoutManagerContract: can't find referenced class com.beloo.widget.chipslayoutmanager.Orientation
Warning:com.beloo.widget.chipslayoutmanager.IStateHolder: can't find referenced class com.beloo.widget.chipslayoutmanager.Orientation
```

I'm trying to keep everything related to `ChipsLayoutManager` with the following annotations:
```
-keep class com.beloo.widget.chipslayoutmanager.* { *; }
-keep class com.beloo.widget.chipslayoutmanager.*$* { *; }
```

**Comment by icarovirtual** [comment]
I made some changes and by using the configuration below I do not get any ProGuard warnings.
```
-keep class com.beloo.widget.chipslayoutmanager.* { *; }
-keep class com.beloo.widget.chipslayoutmanager.** { *; }
-keep class com.beloo.widget.chipslayoutmanager.*$* { *; }
-keep class RestrictTo.*
-keep class RestrictTo.**
-keep class RestrictTo.*$*
```

**Comment by Krisztiaan** [comment]
These are very vague rules. It would be sufficient, if the jCenter library just contained the Orientations.java.
I imported just that file in the project, and it works without any hassle.

**Comment by egistli** [comment]
Rules provided by @ihhcarus are not working on my case.
I use -dontwarn rule to fix this temporarily.
```
-dontwarn com.beloo.widget.chipslayoutmanager.Orientation
```
But do you guys know the cause of this issue? Is it because the Orientation annotation is not kept when building the jar?
There is no `Orientation` annotation in the jar file.
And why there's no problem for building debug variant (of the app includes ChipsLayoutManager)?

**Comment by Krisztiaan** [comment]
One workaround for now, would be adding the Orientation.java to your project yourself, to the correct package.

**Comment by icarovirtual** [comment]
Yes @egistli, what I posted was also not working that well for me. 
I didn't comment back here again but I ended up using `-dontwarn` too to fix this temporarily.

**Comment by egistli** [comment]
@Krisztiaan Add the Orientation.java to the app project will resolve this issue but I prefer to keep third-party packages/files out of the app project and only referenced them via `build.gradle`.

**Comment by galex** [comment]
Same problem here, is there a fix planned in a future version?

**Comment by cristianireyes** [comment]
I got the same problem when trying to sign and build my application. The solution was to add "-dontwarn com.beloo.widget.chipslayoutmanager. **" in my file "proguard-rules.pro"
Once this was done, it worked.

This is the content of my "proguard-rules.pro" file:
`
`#` Add project specific ProGuard rules here.
# By default, the flags in this file are appended to flags specified
# in D:\ANDROID\SDK/tools/proguard/proguard-android.txt
# You can edit the include path and order by changing the proguardFiles
# directive in build.gradle.
#
# For more details, see
#   http://developer.android.com/guide/developing/tools/proguard.html

# Add any project specific keep options here:

# If your project uses WebView with JS, uncomment the following
# and specify the fully qualified class name to the JavaScript interface
# class:
-keepclassmembers class fqcn.of.javascript.interface.for.webview {
   public *;
}

# Uncomment this to preserve the line number information for
# debugging stack traces.
#-keepattributes SourceFile,LineNumberTable

# If you keep the line number information, uncomment this to
# hide the original source file name.
#-renamesourcefileattribute SourceFile

-dontwarn com.beloo.widget.chipslayoutmanager.**
`

**Comment by cristianireyes** [comment]
I got the same problem when trying to sign and build my application. The solution was to add "-dontwarn com.beloo.widget.chipslayoutmanager. **" in my file "proguard-rules.pro"
Once this was done, it worked.

Regards.

**Comment by ajay-aravind** [comment]
@cristianireyes would it affect any functionality?
